spec/models/like_spec.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Like, type: :model do
+  describe 'validations' do
+    let!(:user) { create(:user) }
+    let!(:newpost) { create(:post) }
+
+    it { should validate_presence_of(:user_id) }
+    it { should validate_presence_of(:likable_id) }
+
+    it 'validates uniqueness of likable_id scoped to user_id' do
+      create(:like, user:, likable: newpost)
+      should validate_uniqueness_of(:likable_id).scoped_to(:user_id)
+    end
+
+    it 'validates uniqueness of user_id scoped to likable_id' do
+      create(:like, user:, likable: newpost)
+      should validate_uniqueness_of(:user_id).scoped_to(:likable_id)
+    end
```

Comment:
You don't need to create the user, like and post. Just do the:

`should validate_uniqueness_of(:likable_id).scoped_to(:user_id)`
`should validate_uniqueness_of(:user_id).scoped_to(:likable_id)`